AbstractDffractometer, AbstractSampleView and inheriting classes (#1479)
mxcube/mxcubecore: abstract_diffractometer_breaking_changes.md

> In mxcubecore there are two classes, which could be seen as approximate abstraction for controlling a diffractometer - ***MiniDiff*** and ***GenericDiffractometer***. These classes handle also some sample view and sample centring functionalities. It exists as well an ***AbstractSampleView*** class, which contains only few methods, used to handle the sample visualisation.

Reviewer: ```suggestion
In mxcubecore there are two classes, which could be seen as approximate abstractions for controlling a diffractometer - ***MiniDiff*** and ***GenericDiffractometer***. These classes handle also some sample view and sample centring functionalities. There is also an ***AbstractSampleView*** class, which contains only few methods, used to handle the sample visualisation.
```

The new **AbstractDiffractometer** class and the refactoring of the **AbstractSampleView/SampleView** classes aim to make a standard diffractometer API as well as group all the sample viewing methods to the sample view classes. This intorduces some breaking changes.

#### A convention has been introduced for the motors of the diffractometer.

Thus, the object names change as follows:

- **MiniDiff**:
  - phiMotor - omega
  - sampleXMotor - sampx
  - sampleYMotor - sampy
  - phiyMotor - phiy
  - phixMotor - phiz
  - kappaMotor - kappa
  - kappaPhiMotor - kappa_phi
  - zoomMotor - zoom, as NState, not as motor
- **GenericDiffractometer**:
  - phi - omega

#### Transfer of all the sample view and centring methods to the sample view objects, including the list of the sample centring motors.

This implies that some methods are invoked not as HWR.beamline.diffractometer, but as HWR.beamline.sample_view. Here follows the list of these methods:

- take_snapshot
- get_centring_status
- get_positions
- get_centred_point_from_coord
- motor_positions_to_screen
- current_centring_procedure
- cancel_centring_method
- accept_centring
- reject_centring

> #### The centring motors are defined in to sample_view and not diffractometer any more.

Reviewer: ```suggestion
#### The centring motors are defined in sample_view and not diffractometer any more.
```
